### DISCUSSION TITLE
what is the best way to export github Projects to excel

### DISCUSSION THREAD

**Discussion by Changliu52** [community/community#42844] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

Hi,

I cannot find a way to export the project as list to be able to share it easily for our project review.

Please advice the best way to do this.

Best Regards,
Chang

**Comment by nichita-pasecinic** [comment]
@Changliu52 u could use github API, GET request to list repositories, then convert from JSON to excel format
https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-organization-repositories

**Reply by Changliu52** [reply]
Thanks, but I meant the github projects view, which can register issues as list and Kanban for the purpose of project management.

**Reply by Changliu52** [reply]
we need a way to take snapshot of the project and review with wider team.

**Reply by Changliu52** [reply]
https://github.com/features/issues

**Reply by nichita-pasecinic** [reply]
There is a [graphql api](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects#finding-information-about-items-in-a-project) for that as well

**Comment by kalanchan** [comment]
Also looking for a solution to this, is there a way to flatten the json response so that comments can be returned as a single contacted string instead of an array?

**Reply by oskarpie** [reply]
To flatten JSON you could try [Gron](https://github.com/tomnomnom/gron).

**Reply by kalanchan** [reply]
Been playing around with this for the last 2 weeks, it seems that Projects and Issues don't talk to each other in the Github API. I need some of the columns from Projects but they aren't available from the Issue response body. 

Any ideas?

**Reply by JosipNizetic** [reply]
This would be also useful for me to know. 

Thank you.

**Comment by jmbowman** [comment]
I just discovered https://github.com/justinmahar/github-projectv2-csv-exporter , which isn't perfect but exports a lot of the relevant data with minimal effort.

**Comment by jmbowman** [comment]
Actually, I had even better luck with https://github.com/fiedl/github-project-to-csv .  It exported the descriptions and custom fields that github-projectv2-csv-exporter omitted.

**Reply by Z-K-O** [reply]
After a million and one different solutions, including a bunch of GraphQL shenanigans, this has saved my sanity and finally pulled *all* the data from both the Issue and the connected ProjectV2 Card into a simple semicolon-delimited csv. (Just add sep=; as a new line at the beginning of the csv.) I used the Docker method since I'm on Windows.
